### PR TITLE
Add download feature and rendering for all the signals

### PR DESCRIPTION
**Pull request by WJ0417**

### CODE REVIEW

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  canvas,
+  videoRecord,
+  card2,
+  card4,
+  timerDisplay,
+} from './globals_dom';
+
+import { ffmpegWorkerCombine, ffmpegWorkerAudio } from './globals_worker';
+export function createDownload(inputWidth: number, inputHeight: number): void {
+  // Puases the main preview video player to ensure recording quality
```
Comment (parrajustin) [inline web_autoflip_demo/autoflip_demo_UI/src/download.ts:35]: Puases  -> Pauses?

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  videoPreview,
+  canvas,
+  videoRecord,
+  card2,
+  card4,
+  timerDisplay,
+} from './globals_dom';
+
+import { ffmpegWorkerCombine, ffmpegWorkerAudio } from './globals_worker';
+export function createDownload(inputWidth: number, inputHeight: number): void {
```
Comment (parrajustin) [inline web_autoflip_demo/autoflip_demo_UI/src/download.ts:34]: Add js doc describing what the function does

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        render.height,
+      );
+      stream.getVideoTracks()[0].requestFrame();
+      setTimeout(loop, 35);
+    } else {
+      stopRecording();
+    }
+  }
+
+  function handleDataAvailable(event: BlobEvent): void {
```
Comment (parrajustin) [inline web_autoflip_demo/autoflip_demo_UI/src/download.ts:80]: add small js docs to the functions. (js function comments)

File: web_autoflip_demo/autoflip_demo_UI/test/globals.test.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as globals from '../src/globals';
+
+/*
```
Comment (parrajustin) [inline web_autoflip_demo/autoflip_demo_UI/test/globals.test.ts:18]: remove commented lines?

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts (hunk 3 above)
Comment (WJ0417) [reply web_autoflip_demo/autoflip_demo_UI/src/download.ts]: fixed

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts (hunk 2 above)
Comment (WJ0417) [reply web_autoflip_demo/autoflip_demo_UI/src/download.ts:56]: fixed.

File: web_autoflip_demo/autoflip_demo_UI/src/download.ts (hunk 1 above)
Comment (WJ0417) [reply web_autoflip_demo/autoflip_demo_UI/src/download.ts]: fixed.

File: web_autoflip_demo/autoflip_demo_UI/test/globals.test.ts (hunk 4 above)
Comment (WJ0417) [reply web_autoflip_demo/autoflip_demo_UI/test/globals.test.ts]: fixed.

**Pull request merged**